qiskit_quantuminspire/qi_jobs.py

```diff
@@ -52,16 +77,96 @@ def __init__(
             if isinstance(run_input, QuantumCircuit)
             else [CircuitExecutionData(circuit=circuit) for circuit in run_input]
         )
+        self.program_name = "Program created by SDK"
+        self.batch_job_id: Union[int, None] = None
 
-    def submit(self) -> None:
+    async def submit(self) -> None:
         """Submit the (batch)job to the quantum inspire backend.
 
         Use compute-api-client to call the cjm endpoints in the correct order, to submit the jobs.
         """
-        # Here, we will update the self.circuits_run_data and attach the job ids for each circuit
-        for _ in range(1, 3):
-            pass
-        self.job_id = "999"  # ID of the submitted batch-job
+        options = self.backend().options
+        configuration = config()
+        # call create algorithm
+        async with ApiClient(configuration) as api_client:
+
+            project = await self._create_project(api_client, configuration.auth_settings().team_member_id)
+            batch_job = await self._create_batch_job(api_client, backend_type_id=self.backend().id)
+
+            async def job_run_sequence(
+                in_api_client: ApiClient,
+                in_project: Project,
+                in_batch_job: BatchJob,
+                circuit_data: CircuitExecutionData,
+            ) -> None:
+                algorithm = await self._create_algorithm(in_api_client, in_project.id)
+                commit = await self._create_commit(in_api_client, algorithm.id)
+                file = await self._create_file(in_api_client, commit.id, circuit_data.circuit)
+                job: Job = await self._create_job(
+                    in_api_client, file.id, in_batch_job.id, number_of_shots=options.get("shots", default=1000)
+                )
```

Comment:
Isn't default number of shots a property of backend_type?

Reply:
Indeed. Corrected.